vmm/src/lib.rs

```diff
@@ -1551,10 +1551,24 @@ impl Vmm {
                             self.epoll_context.disable_stdin_event();
                         }
                         Ok(count) => {
+                            // There are 2 serial devices instantiated at this point:
+                            // One lives in LegacyDeviceManager and the other in MMIODeviceManager
+                            // Use the one that is actually plugged in the Guest.
+                            #[cfg(target_arch = "aarch64")]
+                            let serial = self
```

Comment:
What I would do here for more elegance would be this:
Have a helper function (maybe `self.get_stdio_serial()`) that has two implementations, one for x86_64 and one for aarch64. See the example of 
`    fn attach_legacy_devices(&mut self) -> std::result::Result<(), StartMicrovmError> {
`
Additionally, if you decide to use the `get_device` function from the `MMIODeviceManager`, this will become `self.mmio_device_manager.get_device(DeviceType::Serial, "uart")`.

Reply:
>     * PortIODeviceManager should be compiled out for aarch64 (as a consequence of this, you could also need to compile out some more things, like the i8042 usages). I would do a separate commit for this.
> 
>     * Remove the `Option` around the `MMIODeviceManager`, this will also benefit the x86_64 logic since we are always calling `self.attach_virtio_devices()?;` at `start_microvm` no matter what.

I will perform the rename, but outcompiling this on aarch64 will backfire by lots of
`#[cfg(target_arch = "x86_64")]` in different parts of the codebase. 
Also removing the Option depends on also initializing guest memory before instantiating the PortIODeviceManager generating even more code changes.
I think this should be done as a separate refactoring PR with a clear design defined and agreed by all.